### PR TITLE
Add Boilerplate Generator

### PR DESCRIPTION
**Pull request by sebastianbenz**
Preview here: 
https://amp-by-example-staging.appspot.com/boilerplate-generator (soonish)

### CODE REVIEW

**Comment by sebastianbenz** [comment]
Yeah, it should be a radio. This requires a bit more work though as the current data format doesn't support this.

**Comment by sebastianbenz** [comment]
OK, added quite a few changes:

* use radio button for json-ld snippets
* smaller buttons for format selection
* fix formatting (had to switch to handlebars to get this right grrrrr)

**Comment by sebastianbenz** [comment]
PTAL

**Comment by sebastianbenz** [comment]
OK, had to make two more small changes, but that should be it...

**Comment by sebastianbenz** [comment]
Live here: https://amp-by-example-staging.appspot.com/boilerplate-generator/

File: boilerplate-generator/templates/files/serviceworkerJs.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  ];
+  if (OFFLINE_PAGE) {
+    urls.push(OFFLINE_PAGE);
+  }
+  event.waitUntil(
+      caches.open(workbox.core.cacheNames.runtime).then(cache => cache.addAll(urls))
+  );
+});
+
+// Enable navigation preload . This is only necessary if navigation routes are not cached,
```
Comment (kul3r4) [inline boilerplate-generator/templates/files/serviceworkerJs.js]: remove space from "preload ."

**Pull request merged**